### PR TITLE
add test to check for receiving callbacks for invalid data

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connects to ros2/rmw_connext#70

### CODE REVIEW

File: test_communication/test/test_subscription_valid_data.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <iostream>
+#include <string>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <test_communication/msg/u_int32.hpp>
```
Comment (wjwwood) [inline test_communication/test/test_subscription_valid_data.cpp]: It's unrelated to this pr, but I'd say the transformation should not put an underscore between capital letters. `u_int32` is pretty silly looking, it should just be `uint32`. On the other hand, consider `RGBDImage` (obviously a manufactured example), which by my suggestion would be `rgbdimage`, should probably be `rgbd_image`. But even more wrong would be `r_g_b_d_image`. Obviously making the "correct" translation is not straight forward.

**Comment by wjwwood** [comment]
+1

File: test_communication/test/test_subscription_valid_data.cpp (hunk 1 above)
Comment (esteve) [inline test_communication/test/test_subscription_valid_data.cpp]: Agree, `u_int32` looks weird. +1 to not put an underscore between capital letters.

**Comment by esteve** [comment]
+1

**Pull request merged**